src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -639,10 +639,14 @@ static SafeContext getDeclarationValEscape(BoundTypeExpression typeExpression, S
 
             static ParameterSymbol? tryGetThisParameter(MethodSymbol method)
             {
-                if (method.IsExtensionMethod) // Tracked by https://github.com/dotnet/roslyn/issues/76130: Test this code path with new extensions
+                if (method.IsExtensionMethod)
                 {
                     return method.Parameters is [{ } firstParameter, ..] ? firstParameter : null;
                 }
+                else if (method.GetIsNewExtensionMember())
+                {
+                    return method.ContainingType.ExtensionParameter;
```

Comment:
>return method.ContainingType.ExtensionParameter; [](http://example.com/codeflow?start=20&length=48)

It is not clear why do we want to have this parameter at all. How does this align with the language spec?

Reply:
I'm not sure I understand the question.
From a ref analysis perspective, it should be equivalent to deconstruct via invocation, deconstruction or recursive pattern. So all three need to account for `scoped` on the receiver parameter. Here's we're looking at a skeleton signature, so this is the parameter we need to check for presence or absence of `scoped`. See `Deconstruction_ScopedRef_ExtensionMethod`.
I can adjust the name of the local function (maybe "tryGetReceiverParameter") if that's what bothers you.